azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -31,18 +31,18 @@ class FileDescriptor:
     GZIP_MAX_DISK_SIZE_FOR_DETECTION = int(4 * 1024 * 1024 * 1024 / 40)
     DEFAULT_COMPRESSION_RATIO = 11
 
-    def __init__(self, path, size=0, source_id=None):
+    def __init__(self, path: str, size: Optional[int] = None, source_id: OptionalUUID = None):
         """
         :param path: file path.
         :type path: str.
         :param size: estimated size of file if known. if None or 0 will try to guess.
         :type size: int.
-        :param source_id uuid: a v4 uuid to serve as the sources id.
-        :type source_id: str (of a uuid4) or uuid4.
+        :param source_id: a v4 uuid to serve as the sources id.
+        :type source_id: OptionalUUID
         """
         self.path = path
         self._size = size
-        self._detect_size_once = size < 1
+        self._detect_size_once = not size
```

Comment:
That's a breaking change as 0 size (the previous default) would be treated as no detection.
Why not make the blob default 0 instead of None?

Reply:
not 0 == True

The behavior doesn't change